### ISSUE TITLE
monster hunter world not launching

### ISSUE THREAD

**Issue by testc2n14** [ValveSoftware/gamescope#1280]
i've used very reasource i coudl and used the xamples and help section to try and figure it out but no matter what if i am using game scope with monster hutner world then it can not launch

this is what i have set right now "gamescope -w 3200 -h 1800 -W 3480 -H 2160 --hdr-enabled -F fsr -- %command%"

**Comment by sharkautarch** [comment]
> i've used very reasource i coudl and used the xamples and help section to try and figure it out but no matter what if i am using game scope with monster hutner world then it can not launch
> 
> this is what i have set right now "gamescope -w 3200 -h 1800 -W 3480 -H 2160 --hdr-enabled -F fsr -- %command%"


try launching steam from the command line, so that you can see what gamescope prints out on the command line
Also, what version of gamescope are you using?

**Comment by Joshua-Ashton** [comment]
Please try latest tag

**Comment by testc2n14** [comment]
> > i've used very reasource i coudl and used the xamples and help section to try and figure it out but no matter what if i am using game scope with monster hutner world then it can not launch
> > this is what i have set right now "gamescope -w 3200 -h 1800 -W 3480 -H 2160 --hdr-enabled -F fsr -- %command%"
> 
> try launching steam from the command line, so that you can see what gamescope prints out on the command line Also, what version of gamescope are you using?

how do i launch it from terminal

**Comment by sharkautarch** [comment]
> how do i launch it from terminal


You can launch steam from the terminal by opening a terminal emulator window (on KDE it's called "konsole", or you could try pressing `ctrl`+`alt` + `t`) then typing and entering `steam`

@testc2n14 
Also Joshua-Ashton said you should try using the latest release tag of gamescope. On archlinux, you can do that by installing gamescope-git,
~~on other distros, you can just download the source directly: https://github.com/ValveSoftware/gamescope/releases/tag/3.14.12~~
on other distros, you can just git clone the gamescope repo, and then checkout on the latest release tag: 
First, make sure you have the `git`, `meson`, `gcc`, (also on ubuntu: `ninja-build`, `pkg-config`)  installed

 (EDIT: @Joshua-Ashton I just realized there's something weird with the latest release's source downloads: fallback is forced for wlroots and vkroots, BUT there's no wrap file for either, and the source doesn't have a `.gitmodules` file. Probably not much of a problem when being used w/ a distro's packaging system, but inconvenient for people wanting to directly build from the source)




all commands you'll need to run:
```
mkdir ~/gamescope-pkg
cd ~/gamescope-pkg
git clone https://github.com/ValveSoftware/gamescope.git
cd gamescope
git checkout 3.14.12
meson subprojects download
git submodule update
meson setup build -Dinput_emulation=false --reconfigure --wipe --buildtype=debugoptimized -Db_lto=true -Denable_openvr_support=false -Dc_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dc_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer"
ninja -C build
```
Now to run this newer gamescope executable, you just replace `gamescope` in steam launcher with `${HOME}/gamescope-pkg/gamescope/build/src/gamescope`

**Comment by testc2n14** [comment]
> > how do i launch it from terminal
> 
> You can launch steam from the terminal by opening a terminal emulator window (on KDE it's called "konsole", or you could try pressing `ctrl`+`alt` + `t`) then typing and entering `steam`
> 
> @testc2n14 Also Joshua-Ashton said you should try using the latest release tag of gamescope. On archlinux, you can do that by installing gamescope-git, ~on other distros, you can just download the source directly: https://github.com/ValveSoftware/gamescope/releases/tag/3.14.12~ on other distros, you can just git clone the gamescope repo, and then checkout on the latest release tag: First, make sure you have the `git`, `meson`, `gcc`, (also on ubuntu: `ninja-build`, `pkg-config`) installed
> 
> (EDIT: @Joshua-Ashton I just realized there's something weird with the latest release's source downloads: fallback is forced for wlroots and vkroots, BUT there's no wrap file for either, and the source doesn't have a `.gitmodules` file. Probably not much of a problem when being used w/ a distro's packaging system, but inconvenient for people wanting to directly build from the source)
> 
> all commands you'll need to run:
> 
> ```
> mkdir ~/gamescope-pkg
> cd ~/gamescope-pkg
> git clone https://github.com/ValveSoftware/gamescope.git
> cd gamescope
> git checkout 3.14.12
> meson subprojects download
> git submodule update
> meson setup build -Dinput_emulation=false --reconfigure --wipe --buildtype=debugoptimized -Db_lto=true -Denable_openvr_support=false -Dc_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dc_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer"
> ninja -C build
> ```
> 
> Now to run this newer gamescope executable, you just replace `gamescope` in steam launcher with `${HOME}/gamescope-pkg/gamescope/build/src/gamescope`

i ran the command but i got this error, dose sudo dnf install gamescope verion work, also what is the eqivlents to discords ```something``` to make the code block apperar.  this was the error

meson.build:1:0: ERROR: Value "false" (of type "string") for combo option "Support for XTest/Input Emulation with libei" is not one of the choices. Possible choices are (as string): "enabled", "disabled", "auto".

A full log can be found at /home/c2n14/gamescope-pkg/gamescope/build/meson-logs/meson-log.txt
ninja: Entering directory `build'
ninja: error: loading 'build.ninja': No such file or directory

**Comment by sharkautarch** [comment]
> > Replying to [#1280 (comment)](https://github.com/ValveSoftware/gamescope/issues/1280#issuecomment-2094326724)
> 
> i ran the command but i got this error, dose sudo dnf install gamescope verion work, also what is the eqivlents to discords `something` to make the code block apperar. this was the error
> 
> meson.build:1:0: ERROR: Value "false" (of type "string") for combo option "Support for XTest/Input Emulation with libei" is not one of the choices. Possible choices are (as string): "enabled", "disabled", "auto".
> 
> A full log can be found at /home/c2n14/gamescope-pkg/gamescope/build/meson-logs/meson-log.txt ninja: Entering directory `build' ninja: error: loading 'build.ninja': No such file or directory


oh I got part of the command wrong
Here's the revised version:
```
mkdir ~/gamescope-pkg
cd ~/gamescope-pkg
git clone https://github.com/ValveSoftware/gamescope.git
cd gamescope
git checkout 3.14.12
meson subprojects download
git submodule update
meson setup build -Dinput_emulation=disabled --reconfigure --wipe --buildtype=debugoptimized -Db_lto=true -Denable_openvr_support=false -Dc_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dc_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer"
ninja -C build
```

**Comment by testc2n14** [comment]
> > > Replying to [#1280 (comment)](https://github.com/ValveSoftware/gamescope/issues/1280#issuecomment-2094326724)
> > 
> > 
> > i ran the command but i got this error, dose sudo dnf install gamescope verion work, also what is the eqivlents to discords `something` to make the code block apperar. this was the error
> > meson.build:1:0: ERROR: Value "false" (of type "string") for combo option "Support for XTest/Input Emulation with libei" is not one of the choices. Possible choices are (as string): "enabled", "disabled", "auto".
> > A full log can be found at /home/c2n14/gamescope-pkg/gamescope/build/meson-logs/meson-log.txt ninja: Entering directory `build' ninja: error: loading 'build.ninja': No such file or directory
> 
> oh I got part of the command wrong Here's the revised version:
> 
> ```
> mkdir ~/gamescope-pkg
> cd ~/gamescope-pkg
> git clone https://github.com/ValveSoftware/gamescope.git
> cd gamescope
> git checkout 3.14.12
> meson subprojects download
> git submodule update
> meson setup build -Dinput_emulation=disabled --reconfigure --wipe --buildtype=debugoptimized -Db_lto=true -Denable_openvr_support=false -Dc_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dc_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer"
> ninja -C build
> ```

so i ran that and got this error, can i jsut downl;aod the flatpac isntead using sudo dnf install gamescope, that is what i was using befor.     

Run-time dependency libdecor-0 found: NO (tried pkgconfig and cmake)

src/meson.build:41:15: ERROR: Dependency "libdecor-0" not found, tried pkgconfig and cmake

A full log can be found at /home/c2n14/gamescope-pkg/gamescope/build/meson-logs/meson-log.txt
ninja: Entering directory `build'
ninja: error: loading 'build.ninja': No such file or directory

**Comment by sharkautarch** [comment]
> > [#1280 (comment)](https://github.com/ValveSoftware/gamescope/issues/1280#issuecomment-2094786504)
> 
> so i ran that and got this error, can i jsut downl;aod the flatpac isntead using sudo dnf install gamescope, that is what i was using befor.
> 
> Run-time dependency libdecor-0 found: NO (tried pkgconfig and cmake)
> 
> src/meson.build:41:15: ERROR: Dependency "libdecor-0" not found, tried pkgconfig and cmake
> 
> A full log can be found at /home/c2n14/gamescope-pkg/gamescope/build/meson-logs/meson-log.txt ninja: Entering directory `build' ninja: error: loading 'build.ninja': No such file or directory


Sure you can try installing and using gamescope in flatpak
If you don't have any issues with the flatpak version, then that'll probably be the easiest solution for you

**Comment by testc2n14** [comment]
> > Replying to [#1280 (comment)](https://github.com/ValveSoftware/gamescope/issues/1280#issuecomment-2094799244)
> 
> Sure you can try installing and using gamescope in flatpak If you don't have any issues with the flatpak version, then that'll probably be the easiest solution for you
> 
> I believe that the current version of the gamescope flatpak is `org.freedesktop.Platform.VulkanLayer.gamescope`, is that what you have installed? (should be able to check by running `flatpak list`)

yeah it;s ont he newest version but i stil get crashes when running that command whta do i do then

**Comment by sharkautarch** [comment]
> > > Replying to [#1280 (comment)](https://github.com/ValveSoftware/gamescope/issues/1280#issuecomment-2094799244)
> > 
> > 
> > Sure you can try installing and using gamescope in flatpak If you don't have any issues with the flatpak version, then that'll probably be the easiest solution for you
> > I believe that the current version of the gamescope flatpak is `org.freedesktop.Platform.VulkanLayer.gamescope`, is that what you have installed? (should be able to check by running `flatpak list`)
> 
> yeah it;s ont he newest version but i stil get crashes when running that command whta do i do then


In that case:
Check if it runs if you run `ENABLE_GAMESCOPE_WSI=0 gamescope …`

if it still doesn’t run, then you can build an older version:
```sh
[ -e ~/gamescope-pkg ] && rm -rf "${HOME}/gamescope-pkg"
mkdir ~/gamescope-pkg
cd ~/gamescope-pkg
git clone https://github.com/ValveSoftware/gamescope.git
cd gamescope
git checkout 14a1db3
meson subprojects download
git submodule update
meson setup build --reconfigure --wipe --buildtype=debugoptimized -Db_lto=true -Denable_openvr_support=false -Dc_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dc_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer"
ninja -C build
```

**Comment by testc2n14** [comment]
> > > > Replying to [#1280 (comment)](https://github.com/ValveSoftware/gamescope/issues/1280#issuecomment-2094799244)
> > > 
> > > 
> > > Sure you can try installing and using gamescope in flatpak If you don't have any issues with the flatpak version, then that'll probably be the easiest solution for you
> > > I believe that the current version of the gamescope flatpak is `org.freedesktop.Platform.VulkanLayer.gamescope`, is that what you have installed? (should be able to check by running `flatpak list`)
> > 
> > 
> > yeah it;s ont he newest version but i stil get crashes when running that command whta do i do then
> 
> In that case: Check if it runs if you run `ENABLE_GAMESCOPE_WSI=0 gamescope …`
> 
> if it still doesn’t run, then you can build an older version:
> 
> ```shell
> [ -e ~/gamescope-pkg ] && rm -rf "${HOME}/gamescope-pkg"
> mkdir ~/gamescope-pkg
> cd ~/gamescope-pkg
> git clone https://github.com/ValveSoftware/gamescope.git
> cd gamescope
> git checkout 14a1db3
> meson subprojects download
> git submodule update
> meson setup build --reconfigure --wipe --buildtype=debugoptimized -Db_lto=true -Denable_openvr_support=false -Dc_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dc_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer" -Dcpp_link_args="-g1 -Wno-error=unused-but-set-variable -Wno-error=unused-variable -fno-omit-frame-pointer"
> ninja -C build
> ```

so i ran that got alot fo wanring and a few errors

c2n14@fedora:~$ ENABLE_GAMESCOPE_WSI=0 gamescope …
No CAP_SYS_NICE, falling back to regular-priority compute and threads.
Performance will be affected.
ATTENTION: default value of option vk_khr_present_wait overridden by environment.
ATTENTION: default value of option vk_khr_present_wait overridden by environment.
ATTENTION: default value of option vk_khr_present_wait overridden by environment.
ATTENTION: default value of option vk_khr_present_wait overridden by environment.
vulkan: selecting physical device 'AMD Radeon RX 6800 (RADV NAVI21)': queue family 1 (general queue family 0)
vulkan: physical device supports DRM format modifiers
wlserver: [backend/headless/backend.c:67] Creating headless backend
vulkan: supported DRM formats for sampling usage:
vulkan:   AR24 (0x34325241)
vulkan:   XR24 (0x34325258)
vulkan:   AB24 (0x34324241)
vulkan:   XB24 (0x34324258)
vulkan:   RG16 (0x36314752)
vulkan:   NV12 (0x3231564E)
vulkan:   AB4H (0x48344241)
vulkan:   XB4H (0x48344258)
vulkan:   AB48 (0x38344241)
vulkan:   XB48 (0x38344258)
vulkan:   AB30 (0x30334241)
vulkan:   XB30 (0x30334258)
vulkan:   AR30 (0x30335241)
vulkan:   XR30 (0x30335258)
vulkan: Creating Gamescope nested swapchain with format 64 and colorspace 0
wlserver: Running compositor on wayland display 'gamescope-0'
wlserver: [backend/headless/backend.c:17] Starting headless backend
wlserver: [xwayland/server.c:108] Starting Xwayland on :1
The XKEYBOARD keymap compiler (xkbcomp) reports:
> Warning:          Could not resolve keysym XF86CameraAccessEnable
> Warning:          Could not resolve keysym XF86CameraAccessDisable
> Warning:          Could not resolve keysym XF86CameraAccessToggle
> Warning:          Could not resolve keysym XF86NextElement
> Warning:          Could not resolve keysym XF86PreviousElement
> Warning:          Could not resolve keysym XF86AutopilotEngageToggle
> Warning:          Could not resolve keysym XF86MarkWaypoint
> Warning:          Could not resolve keysym XF86Sos
> Warning:          Could not resolve keysym XF86NavChart
> Warning:          Could not resolve keysym XF86FishingChart
> Warning:          Could not resolve keysym XF86SingleRangeRadar
> Warning:          Could not resolve keysym XF86DualRangeRadar
> Warning:          Could not resolve keysym XF86RadarOverlay
> Warning:          Could not resolve keysym XF86TraditionalSonar
> Warning:          Could not resolve keysym XF86ClearvuSonar
> Warning:          Could not resolve keysym XF86SidevuSonar
> Warning:          Could not resolve keysym XF86NavInfo
Errors from xkbcomp are not fatal to the X server
wlserver: [types/wlr_compositor.c:692] New wlr_surface 0x5582eb509f40 (res 0x5582eb5103c0)
wlserver: [xwayland/server.c:273] Xserver is ready
pipewire: stream state changed: connecting
pipewire: stream state changed: paused
pipewire: stream available on node ID: 107
vblank: xwm: execvp failed: No such file or directory
Using timerfd.
gamescope: children shut down!
(EE) failed to read Wayland events: Broken pipe
c2n14@fedora:~$

**Comment by sharkautarch** [comment]
@testc2n14 
> vblank: xwm: execvp failed: No such file or directory

ok to clarify, when I said to run `ENABLE_GAMESCOPE_WSI=0 gamescope …`
I was implying that you were supposed to replace `...` with the other parameters you were using before: `-w 3200 -h 1800 -W 3480 -H 2160 --hdr-enabled -F fsr -- %command%`

Sorry for the confusion

**Comment by testc2n14** [comment]
ahhh ok, yeah so put that intot eh game prorpters nto terminal kk ty

**Comment by testc2n14** [comment]
so i got the game too boot with thsoe comands and int eh taskbar it shows the wayland logo isntead of the normal game logo and wasn;t enabled so i am not sure it is running. i tried launching it a second time it was taking a while so i stoped the launch and tried it again. this crashed steam. now the game is jsut hangiong ont he launching prosses

**Comment by sharkautarch** [comment]
> i know i seem pretty stupid but i am only jsut startign to daily drive linux on my main machine so i still figureing things out

No worries, it's nothing to feel stupid about, I wouldn't expect anyone to be a command-line guru just because they are a linux user :)


> so i got the game too boot with thsoe comands and int eh taskbar it shows the wayland logo isntead of the normal game logo and wasn;t enabled so i am not sure it is running. i tried launching it a second time it was taking a while so i stoped the launch and tried it again. this crashed steam. now the game is jsut hangiong ont he launching prosses

Hmm, would be helpful to see the stuff gamescope is printing to console
Open a terminal by doing `ctrl` + `alt` + `t`,
then enter `steam &| tee gamescope.log`
then launch the game again, and upload the `gamescope.log` you get to gist.github.com and post a link to the gist here

Let me know if you need any extra help with troubleshooting

**Comment by testc2n14** [comment]
gamer laucnhed perfectly fin but as far as i can tell game scope is not running as hdr is not truned on (the game supports hdr on steam deck) https://gist.github.com/testc2n14/dd62f41914cd7c03ea088f0efeb79815#file-gistfile1-txt

**Comment by sharkautarch** [comment]
> gamer laucnhed perfectly fin but as far as i can tell game scope is not running as hdr is not truned on (the game supports hdr on steam deck) https://gist.github.com/testc2n14/dd62f41914cd7c03ea088f0efeb79815#file-gistfile1-txt


> (process:6673): GLib-GObject-CRITICAL **: 19:23:39.694: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
reaping pid: 6673 -- gameoverlayui


hmmm I think there's currently some bug with running games on gamescope when the steam overlay thingy is on
Try launching the game but with steam overlay turned off in the steam options for that game

**Comment by testc2n14** [comment]
by steam voeraly you mean the thingy that pops up when you press alt tab right

**Comment by testc2n14** [comment]
also it;s not enabled

**Comment by testc2n14** [comment]
or atl;east steam says it;s nto for all of my games

**Comment by sharkautarch** [comment]
> or atl;east steam says it;s nto for all of my games


hmmm
try using: `ENABLE_GAMESCOPE_WSI=0 LD_PRELOAD="" gamescope -w 3200 -h 1800 -W 3480 -H 2160 --hdr-enabled -F fsr -- %command%`

**Comment by testc2n14** [comment]
yeah no dice https://gist.github.com/testc2n14/9c765c46d54210f32f0c3a96fcbb9894#file-gistfile1-txt

**Comment by sharkautarch** [comment]
> yeah no dice https://gist.github.com/testc2n14/9c765c46d54210f32f0c3a96fcbb9894#file-gistfile1-txt


hmmm
yeah uh the only other idea I have is for you to look and see if downgrading gamescope helps:
https://koji.fedoraproject.org/koji/packageinfo?packageID=32706

**Comment by testc2n14** [comment]
well thx for the help. well atleast it;s good know that it's abnormal behavoirs. could there be sokething my install of steam is doing thats wrong or something weird with this game

**Comment by testc2n14** [comment]
also how is gamescope implemented into steam deck

**Comment by sharkautarch** [comment]
> well thx for the help. well atleast it;s good know that it's abnormal behavoirs. could there be sokething my install of steam is doing thats wrong or something weird with this game

Yeah unfortunately, the logs didn't really show any obvious signs of something going wrong inside gamescope itself (besides it not working ofc)

> also how is gamescope implemented into steam deck


The way gamescope works on the steam deck, is that it by default uses 'embedded' mode (basically a mode w/ no/minimal compositing effects, where gamescope runs the whole desktop session the same way that gnome/kde would). 
Steam deck users can also switch to sortof desktop mode wherein gamescope runs ontop of a normal kde desktop (called nested mode)

I guess there's also some custom integrations gamescope has w/ the steam deck (tho some of said integrations have been getting upstreamed into normal linux releases such as some kernel-level color management features)...

(gamescope's embedded mode isn't something that's steamdeck exclusive, you can run embedded gamescope by just switching to a different TTY (usually using `ctrl` + `alt` + `fn` + `fn<num>` where `fn<num>` is one of fn1-fn12 or something) )
